source/fundamentals/authentication.txt

```diff
@@ -0,0 +1,220 @@
+         var mongoClient = new MongoClient(settings);
+
+.. _csharp-scram-sha-256:
+
+SCRAM-SHA-256
+~~~~~~~~~~~~~
+
+``SCRAM-SHA-256`` is a salted challenge-response authentication mechanism (SCRAM)
+that uses your username and password, encrypted with the ``SHA-256`` algorithm,
+to authenticate your user.
+
+You can specify the ``SCRAM-SHA-256`` authentication mechanism with your connection
+string as follow:
+
+.. code-block:: csharp
+
+   var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=SCRAM-SHA-256"); 
+
+.. tip:: Default Mechanism
+
+   MongoDB version 4.0 and later uses ``SCRAM-SHA-256`` as the default 
+   authentication mechanism if the MongoDB server version supports it.
+
+   To learn more on specifying the default mechanism, see :ref:`csharp-authentication-default`.
+
+.. _csharp-scram-sha-1:
+
+SCRAM-SHA-1
+~~~~~~~~~~~
+
+``SCRAM-SHA-1`` is s a salted challenge-response mechanism (SCRAM) that uses 
+your username and password, encrypted with the ``SHA-1`` algorithm, to authenticate 
+your user.
+
+You can specify the ``SCRAM-SHA-1`` authentication mechanism with your connection
+string as follow:
+
+.. code-block:: csharp
+
+   var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=SCRAM-SHA-1");
+
+.. tip:: Default Mechanism
+
+   MongoDB version 4.0 uses ``SCRAM-SHA-1`` as the default authorization mechanism
+   if the server does not support ``SCRAM-SHA-256``.
+
+   To learn more on specifying the default mechanism, see :ref:`csharp-authentication-default`.
+
+.. _csharp-x509:
+
+X.509
+~~~~~
+
+The ``X.509`` authentication mechanism uses :wikipedia:`TLS <Transport_Layer_Security>`
+with ``X.509`` certificates to authenticate your user, identified by the
+distinguished names of your client certificate. When you specify the 
+``X.509`` authentication mechanism, the server authenticates the connection using 
+the subject name of the client certificate.
+
+To learn more about using TLS/SSL, see our :ref:`TLS/SSL guide <csharp-tls>`.
+
+To learn more about ``X.509`` certificates, see the :ref:`X.509 Server Manual Entry <x509-client-authentication>`.
+
+Select the :guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying the ``X.509`` authentication mechanism:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: default-connection-string
+
+      .. code-block:: csharp
+
+         var connectionString = "mongodb://<hostname>/?authMechanism=MONGODB-X509";
+         var settings = MongoClientSettings.FromConnectionString(connectionString);
+
+         settings.useTls = true;
+         settings.SslSettings = new SslSettings
+         {
+            ClientCertificates = new List<X509Certificate>()
+            {
+                new X509Certificate2("<path to X.509 certificate>", "<X.509 certificate password>")
```

Comment:
Not sure `.pfx` format [here](https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/master/fundamentals/connection/tls/#configure-a-client-certificate). I would probably stick using `.p12` extension. cc: @JamesKovacs

Reply:
@mongoKart copying you here since the PR Dima linked is one you're working on

Reply:
Noted. I can update this when I do the Connection page.

Reply:
I would also want to see a note about pem certificates, but it will require working on CSHARP-4336. Maybe we can work on this soon? I will bring it up to the next retro

Reply:
Apparently CSHARP-4336 is not only about checking whether .pem certificates work or no. It will require changes in the driver, then I'm ok with not mentioning `.pem` certificates for now